dDatabase/CDClientDatabase/CDClientTables/CDObjectsTable.cpp

```diff
@@ -57,14 +56,16 @@ const CDObjects& CDObjectsTable::GetByID(uint32_t LOT) {
 
 	auto tableData = query.execQuery();
 	if (tableData.eof()) {
-		entries.insert(std::make_pair(LOT, m_default));
-		return m_default;
+		entries.emplace(LOT, ObjDefault);
+		return ObjDefault;
 	}
 
 	// Now get the data
 	while (!tableData.eof()) {
-		CDObjects entry;
-		entry.id = tableData.getIntField("id", -1);
+		const uint32_t LOT = tableData.getIntField("id", 0);
```

Comment:
LOT is a typedef, not sure how compilers handle a variable having the same name as a typedef

Reply:
Seems the typedef gets shadowed, but I see your point.